cfht2caom2/tests/test_hdf5.py

```diff
@@ -94,33 +93,34 @@ def pytest_generate_tests(metafunc):
 
 @patch('cfht2caom2.metadata.CFHTCache._try_to_append_to_cache')
 @patch('caom2pipe.astro_composable.get_vo_table')
-def test_visitor(vo_mock, cache_mock, test_name, test_config):
+def test_visitor(vo_mock, cache_mock, test_name, test_config, tmp_path, change_test_dir):
     warnings.simplefilter('ignore', category=AstropyUserWarning)
     warnings.simplefilter('ignore', category=FITSFixedWarning)
     vo_mock.side_effect = test_caom_gen_visit._vo_mock
     # cache_mock there so there are no update cache calls - so the tests
     # work without a network connection
+    test_config.change_working_directory(tmp_path.as_posix())
     storage_name = CFHTName(
-        file_name=basename(test_name),
         instrument=metadata.Inst.SITELLE,
         source_names=[test_name],
     )
     file_info = FileInfo(id=storage_name.file_uri, file_type='application/x-hdf5')
-    metadata_reader = Hdf5FileMetadataReader()
-    metadata_reader.working_directory = SITELLE_DIR
-    metadata_reader.set_headers(storage_name)
-    metadata_reader._file_info = {storage_name.file_uri: file_info}
-    test_config.rejected_fqn = '/tmp/rejected.yml'
-    test_observable = Observable(test_config)
+    storage_name.file_info[storage_name.file_uri] = file_info
+    import h5py
```

Comment:
I think that the import can be moved at the file level since in the tests this is not optional.